conans/model/build_info.py

```diff
@@ -41,49 +47,157 @@ def __init__(self):
         self.description = None  # Description of the conan package
         # When package is editable, filter_empty=False, so empty dirs are maintained
         self.filter_empty = True
+        self._components = OrderedDict()
+
+    @property
+    def _sorted_components(self):
```

Comment:
This algorithm can be done simpler with one loop. The condition is: `Are all the deps of the node already in the list? => push node`

Reply:
Very similar to `_order_levels` in the `graph.py`

Reply:
We need also to introduce a loop check and a test.